pthread/pthread.c

```diff
@@ -13,22 +13,182 @@
  * %LICENSE%
  */
 
-#include "pthread.h"
+#include <stdlib.h>
+#include <errno.h>
+#include <limits.h>
+#include <sys/list.h>
+#include <sys/mman.h>
+#include <sys/minmax.h>
+#include <pthread.h>
 
 
-int pthread_create(pthread_t *thread, const pthread_attr_t *attr, void *(*start_routine)(void *), void *arg)
+#define CEIL(value, size)	((((value) + (size) - 1) / (size)) * (size))
+
+
+static pthread_t *pthread_list = NULL;
+
+
+static const pthread_attr_t pthread_attr_default = {
+	.stackaddr = (void *)0,
+	.policy = SCHED_RR,
+	.priority = 0,
+	.detached = PTHREAD_CREATE_JOINABLE,
+	.stacksize = CEIL(PTHREAD_STACK_MIN, PAGE_SIZE)
+};
+
+
+static void start_point(void *args)
+{
+	pthread_t *thread = (pthread_t *)args;
+
+	thread->retval = (void *)(thread->start_routine(thread->arg));
+
+	endthread();
+}
+
+
+static pthread_t *find_pthread(handle_t id)
```

Comment:
There are cyclic list management functions in our library. Use them, if possible